src/Mvc/Mvc.Core/src/Infrastructure/ResourceInvoker.cs

```diff
@@ -187,11 +143,77 @@ static async Task Logged(ResourceInvoker invoker)
             }
         }
 
+        internal Task ReleaseResourcesCore(IDisposable? scope)
+        {
+            Exception? releaseException = null;
+            ValueTask releaseResult = default;
+            try
+            {
+                releaseResult = ReleaseResources();
+            }
+            catch (Exception exception)
+            {
+                releaseException = exception;
+            }
+
+            if (releaseException == null && !releaseResult.IsCompletedSuccessfully)
```

Comment:
Could we move this inside the `try` block? 

```C#
try
{
   var releaseResult = ReleaseResources();
   if (!releaseResult.IsCompletedSuccessfully)
   {
      return ReleaseResourcesAwaited(releaseResult);
   }
}
catch
```

Reply:
done